great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+import copy
+import logging
+from typing import Any, Dict, List, Optional
+
+from great_expectations.data_context.types.base import BaseYamlConfig
+from great_expectations.marshmallow__shade import INCLUDE, Schema, fields, post_load
+from great_expectations.marshmallow__shade.decorators import post_dump
+from great_expectations.types import DictDot
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DomainBuilderConfig(DictDot):
+    def __init__(
+        self,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ):
+        self._class_name = class_name
+        self._module_name = module_name
+        self.batch_request = batch_request
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during DomainBuilderConfig instantiation', k, v
+            )
+
+    @property
+    def class_name(self) -> str:
+        return self._class_name
+
+    @property
+    def module_name(self) -> Optional[str]:
+        return self._module_name
+
+
+class DomainBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.domain_builder",
+    )
+    batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> DomainBuilderConfig:
+        return DomainBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
```

Comment:
@cdkini If we are going to do this, why not use the library method from top-level `util.py` (e.g., `deep_filter_properties_iterable()` or `filter_properties_dict()` whichever is more appropriate -- perhaps the former)?

Reply:
aha this is why I ask you for reviews! I didn't know about that method let me see how to use it here.

Reply:
@cdkini My guess is that the `deep_filter_properties_iterable()` will work better, but you will know for sure once you do it.  We have many examples in the codebase.

Reply:
I've stuck with filter_properties_dict since we only care about the top level removal of nulls (if a nested object has a null, we should not interfere).